examples/knative/powershell/kn-ps-webhook/README.md

```diff
@@ -0,0 +1,175 @@
+
+![](screenshots/screenshot-01.png)
+
+## Prerequisites
+
+* VEBA deployed with a [wildcard DNS entry](https://en.wikipedia.org/wiki/Wildcard_DNS_record), which is required for the ability to externally invoke a Knative Service URL that is serving the custom webhook function. For more information, please [refer to this blog post](https://williamlam.com/2021/09/custom-webhook-function-to-publish-events-into-vmware-event-broker-appliance-veba.html) for more information.
+
+
+# Step 1 - Build
+
+Create the container image locally to test your function logic.
+
+```console
+export TAG=<version>
+docker build -t <docker-username>/kn-ps-webhook:${TAG} .
+```
+
+# Step 2 - Test
+
+Verify the container image works by executing it locally.
+
+Change into the `test` directory
+```console
+cd test
+```
+
+Update the following variable names within the `docker-test-env-variable` file and either the `send-cloudevent-test.ps1` (PowerShell Script) or `send-cloudevent-test.sh` (Bash Script)
+
+* WEBHOOK_USERNAME - Username to login to webhook function endpoint
+* WEBHOOK_PASSWORD - Password to login to webhook function endpoint
+
+
+Start the container image by running the following command:
+
+```console
+docker run -e FUNCTION_DEBUG=true -e PORT=8080 --env-file docker-test-env-variable -it --rm -p 8080:8080 <docker-username>/kn-ps-webhook:${TAG}
+```
+
+In a separate terminal, run either `send-cloudevent-test.ps1` (PowerShell Script) or `send-cloudevent-test.sh` (Bash Script) to simulate a CloudEvent payload being sent to the local container image
+
+```console
+Testing Function ...
+See docker container console for output
+
+# Output from docker container console
+Id     Name            PSJobTypeName   State         HasMoreData     Location             Command
+--     ----            -------------   -----         -----------     --------             -------
+2      Job2            ThreadJob       NotStarted    False           PowerShell           …
+09/19/2021 13:34:04 - PowerShell HTTP server start listening on 'http://*:8080/'
+09/19/2021 13:34:04 - Processing Init
+
+09/19/2021 13:34:04 - Init Processing Completed
+
+09/19/2021 13:34:09 - Processing Context ...
+
+09/19/2021 13:34:09 - [Function Secrets]:
+{"WEBHOOK_USERNAME": "XXX","WEBHOOK_PASSWORD": "XXX"}
+
+09/19/2021 13:34:09 - Checking for authorization ...
+
+09/19/2021 13:34:09 - [Expected Authorization Header]:
+Basic XXX
+
+09/19/2021 13:34:09 - [Headers]:
+
+Name                           Value
+----                           -----
+Content-Length                 {57}
+User-Agent                     {curl/7.64.1}
+Accept                         {*/*}
+SendToBroker                   {False}
+Host                           {localhost:8080}
+Authorization                  {Basic XXX}
+Content-Type                   {application/json}
+
+09/19/2021 13:34:09 - [Body]:
+@{VmName=Production-01; UserName=Michael Gasch}
+
+09/19/2021 13:34:09 - Creating custom CloudEvent ...
```

Comment:
nit: saying "custom" CloudEvent feels redundant, i.e. "CloudEvent" is just fine (see other occurrences).